features/consent/src/main/java/in/projecteka/jataayu/consent/viewmodel/GrantedConsentListViewModel.kt

```diff
@@ -0,0 +1,84 @@
+package `in`.projecteka.jataayu.consent.viewmodel
+
+import `in`.projecteka.jataayu.consent.R
+import `in`.projecteka.jataayu.consent.extension.grantedConsentList
+import `in`.projecteka.jataayu.consent.model.ConsentFlow
+import `in`.projecteka.jataayu.consent.model.ConsentsListResponse
+import `in`.projecteka.jataayu.consent.model.RevokeConsentRequest
+import `in`.projecteka.jataayu.consent.repository.ConsentRepository
+import `in`.projecteka.jataayu.core.model.Consent
+import `in`.projecteka.jataayu.core.model.RequestStatus
+import `in`.projecteka.jataayu.core.model.RequestStatus.DENIED
+import `in`.projecteka.jataayu.core.model.RequestStatus.GRANTED
+import `in`.projecteka.jataayu.core.model.grantedconsent.GrantedConsentDetailsResponse
+import `in`.projecteka.jataayu.network.utils.PayloadLiveData
+import `in`.projecteka.jataayu.network.utils.fetch
+import `in`.projecteka.jataayu.util.extension.EMPTY
+import `in`.projecteka.jataayu.util.ui.DateTimeUtils
+import android.content.res.Resources
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+
+class GrantedConsentListViewModel(private val repository: ConsentRepository) : ViewModel() {
+
+    val consentListResponse = PayloadLiveData<ConsentsListResponse>()
+    val grantedConsentDetailsResponse = PayloadLiveData<List<GrantedConsentDetailsResponse>>()
+    val revokeConsentResponse = PayloadLiveData<Void>()
+    val grantedConsentsList = MutableLiveData<List<Consent>>()
+
+
+    private val grantedConsentStatusList = listOf(
+        R.string.status_all_granted_consents,
+        R.string.status_active_granted_consents,
+        R.string.status_expired_granted_consents
+    )
+
+    internal var selectedProviderName = String.EMPTY
+
+    fun getConsents() =
+        consentListResponse.fetch(repository.getConsents())
+
+    fun getGrantedConsentDetails(requestId: String) {
+        grantedConsentDetailsResponse.fetch(repository.getGrantedConsentDetails(requestId))
+    }
+
+    fun getConsentRepository(): ConsentRepository = repository
+
+    fun populateFilterItems(resources: Resources, flow: ConsentFlow?): List<String> =
+        grantedConsentStatusList.map { getFormattedItem(resources,it, GRANTED) }
+
+
+    private fun getFormattedItem(
```

Comment:
fragment view model should only do stuff related to granted consents. all other logic, may be deemed unnecessary and can be be removed/moved to concerned VM

Reply:
Can you please mention which code